packages/components/src/icon/index.js

```diff
@@ -17,6 +17,10 @@ function Icon( { icon = null, size, ...additionalProps } ) {
 		return <Dashicon icon={ icon } size={ iconSize } { ...additionalProps } />;
 	}
 
+	if ( Dashicon === icon.type ) {
```

Comment:
This `Icon` vs `Dashicon` usage is confusing, but this should work and make it simpler in the future 👍

Reply:
Yeah, still need to think how to make it work nicely with css classes, currently I'm just copying the classes of the Button parent into the Icon so we can [customize it at the svg primitive level](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/primitives/svg/index.native.js#L20), which is not ideal.

Reply:
This change was messing up with many unit tests, so I partially rolled it back.
I understand the idea of the change but I thought it was better to comply with the tests.

Reply:
Nvm. I found out why the test errors were so spread out. It was some import error.
Rolled back to keep these changes fixing those unit tests.